From a9fa02df3a7ea7798c14ad0c2ebfbd7374b8aad0 Mon Sep 17 00:00:00 2001
From: Joseph Guo <qijian.guo@nxp.com>
Date: Thu, 25 Jul 2024 16:48:01 +0800
Subject: [PATCH 09/19] arm64: dts: add imx93-11x11-frdm-ov5640 dts

Add ov5640 node.
Use XRPi-CAM-MiniSAS convert board to enable the MiniSAS MIPI CSI camera.

Signed-off-by: Joseph Guo <qijian.guo@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |  2 +-
 .../dts/freescale/imx93-11x11-frdm-ov5640.dts | 70 +++++++++++++++++++
 2 files changed, 71 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx93-11x11-frdm-ov5640.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 9c243f7798be..4b4a83e49840 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -386,7 +386,7 @@ dtb-$(CONFIG_ARCH_MXC) += imx93-11x11-evk.dtb \
 			  imx93-11x11-evk-rpmsg.dtb imx93-11x11-evk-rpmsg-lpv.dtb \
 			  imx93-11x11-frdm.dtb imx93-11x11-frdm-mt9m114.dtb \
 			  imx93-11x11-frdm-dsi.dtb imx93-11x11-frdm-tianma-wvga-panel.dtb \
-			  imx93-11x11-frdm-aud-hat.dtb
+			  imx93-11x11-frdm-aud-hat.dtb imx93-11x11-frdm-ov5640.dtb
 
 dtb-$(CONFIG_ARCH_MXC) += imx91-11x11-evk.dtb \
 			  imx91-11x11-evk-flexspi-m2.dtb imx91-11x11-evk-flexspi-nand-m2.dtb \
diff --git a/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-ov5640.dts b/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-ov5640.dts
new file mode 100644
index 000000000000..2468987e3883
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-ov5640.dts
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2024 NXP
+ */
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/pwm/pwm.h>
+#include "imx93-11x11-frdm.dts"
+
+/ {
+	clk_dummy: clock-dummy {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <27000000>;
+		clock-output-names = "clk_dummy";
+	};
+};
+
+&isi_0 {
+	status = "okay";
+
+	cap_device {
+		status = "okay";
+	};
+};
+
+&mipi_csi {
+	status = "okay";
+
+	/* Camera 0  MIPI CSI-2 (CSIS0) */
+	port {
+		mipi_csi0_ep: endpoint {
+			remote-endpoint = <&ov5640_mipi_ep>;
+			cfg-clk-range = <28>;
+			hs-clk-range = <0x7>;
+			data-lanes = <2>;
+		};
+	};
+};
+
+&lpi2c3 {
+	/delete-node/ap1302_mipi@3c;
+	ov5640_mipi: ov5640_mipi@3c {
+		compatible = "ovti,ov5640";
+		reg = <0x3c>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_csi_pwdn>;
+		clocks = <&clk_dummy>;
+		clock-names = "xclk";
+		csi_id = <0>;
+		reset-gpios = <&pcal6524 22 GPIO_ACTIVE_LOW>;
+		powerdown-gpios = <&gpio4 28 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+		port {
+			ov5640_mipi_ep: endpoint {
+				remote-endpoint = <&mipi_csi0_ep>;
+				data-lanes = <1 2>;
+				clock-lanes = <0>;
+			};
+		};
+	};
+};
+
+&iomuxc {
+	pinctrl_csi_pwdn: csi_pwdn_grp{
+		fsl,pins = <
+			MX93_PAD_CCM_CLKO3__GPIO4_IO28	0x31e
+		>;
+	};
+};
-- 
2.25.1

